schemas/bolt-yaml-plan.schema.json

```diff
@@ -0,0 +1,294 @@
+    },
+    "steps": {
+      "description": "A list of steps that describe specific actions to take.",
+      "type": "array",
+      "items": {
+        "oneOf": [
+          {
+            "$ref": "#/steps/command"
+          },
+          {
+            "$ref": "#/steps/download"
+          },
+          {
+            "$ref": "#/steps/message"
+          },
+          {
+            "$ref": "#/steps/plan"
+          },
+          {
+            "$ref": "#/steps/resources"
+          },
+          {
+            "$ref": "#/steps/script"
+          },
+          {
+            "$ref": "#/steps/task"
+          },
+          {
+            "$ref": "#/steps/upload"
+          }
+        ]
+      }
+    }
+  },
+  "steps": {
+    "command": {
+      "description": "Runs a single command on a list of targets and returns the results. Results include output sent to standard out (stdout) and standard error (stderr), and the command's exit code.",
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["command", "targets"],
+      "properties": {
+        "command": {
+          "description": "The command to run.",
+          "type": "string"
+        },
+        "description": {
+          "$ref": "#/definitions/description"
+        },
+        "name": {
+          "$ref": "#/definitions/name"
+        },
+        "targets": {
+          "$ref": "#/definitions/targets"
+        }
+      }
+    },
+    "download": {
+      "description": "Downloads a file or directory from a target to a destination directory on the local host.",
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["destination", "download", "targets"],
+      "properties": {
+        "description": {
+          "$ref": "#/definitions/description"
+        },
+        "destination": {
+          "description": "The destination directory to download the file to. Expands relative to the project's downloads directory <project>/downloads.",
+          "type": "string"
+        },
+        "download": {
+          "description": "The location of the remote file or directory to download.",
+          "type": "string"
+        },
+        "name": {
+          "$ref": "#/definitions/name"
+        },
+        "targets": {
+          "$ref": "#/definitions/targets"
+        }
+      }
+    },
+    "message": {
+      "description": "Print a message. This will print a message to standard out (stdout) when using the 'human' output format or standard error (stderr) when using the 'json' output format.",
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["message"],
+      "properties": {
+        "description": {
+          "$ref": "#/definitions/description"
+        },
+        "message": {
+          "description": "The message to print. Messages will be specially formatted if they are valid PlanResult objects, otherwise they will be stringified."
+        },
+        "name": {
```

Comment:
Can message take a name? What does it return that can be assigned to a variable?

Reply:
Yep. I specifically asked about this during review when this step was added, and we decided to keep it in to maintain consistency between steps. The value will always be `nil`.